File: voltdb/procs/SelectUser.java
import org.voltdb.*;

public class SelectUser extends VoltProcedure {
  public final SQLStmt sql =
    new SQLStmt("SELECT * FROM users WHERE id=? ORDER by id LIMIT 1");

  public VoltTable run(long id) {
    voltQueueSQL(sql, id);
    return voltExecuteSQL()[0];
  }
}

File: voltdb/procs/select_user.clj
(require 'defproc)

(defproc select-user
  [long] org.voltdb.VoltTable
  [this & args]
  {"select" (stmt "SELECT * FROM users WHERE id=? LIMIT 1")}

  (.voltQueueSQL this (get-stmt this "select") (into-array args))
  (nth (.voltExecuteSQL this) 0))

;(println (macroexpand '(defproc select-user)))
